src/lib.js

```diff
@@ -3,130 +3,137 @@ var got = require('got'),
     R = require('ramda'),
     path = require('path'),
     paths = require('./paths'),
-    fs = require('fs'),
     os = require('os'),
     util = require('./util'),
     Promise = require('bluebird'),
+    fs = Promise.promisifyAll(require('fs')),
     PURESCRIPT_REPO_API_URL = 'https://api.github.com/repos/purescript/purescript',
     PURESCRIPT_DOWNLOAD_URL = 'https://github.com/purescript/purescript';
 
-function getReleases () {
-  return gotGithubApi('/releases')
-    .then(util.parseResponseBody)
-    .then(function (body) {
-      return R.map(R.prop('tag_name'), body);
-  });
+function getReleases() {
+    return gotGithubApi('/releases')
+        .then(util.parseResponseBody)
+        .then(function (body) {
+            return R.map(R.prop('tag_name'), body);
+        });
 }
 
-function getLatestRelease () {
-  return gotGithubApi('/releases/latest')
-    .then(util.parseResponseBody)
-    .then(function (body) {
-      return R.prop('tag_name', body);
-    });
+function getLatestRelease() {
+    return gotGithubApi('/releases/latest')
+        .then(util.parseResponseBody)
+        .then(function (body) {
+            return R.prop('tag_name', body);
+        });
 }
 
-function gotGithubApi (path) {
-  return got(PURESCRIPT_REPO_API_URL + path, {
-    headers: process.env.GITHUB_API_TOKEN ?
-      { 'authorization': 'token ' + process.env.GITHUB_API_TOKEN } :
-      {}
-  });
+function gotGithubApi(path) {
+    return got(PURESCRIPT_REPO_API_URL + path, {
+        headers: process.env.GITHUB_API_TOKEN ? {
+            'authorization': 'token ' + process.env.GITHUB_API_TOKEN
+        } : {}
+    });
 }
 
-function getInstalledVersions () {
-  var dirs = fs.readdirSync(paths.PSVM_VERSIONS);
-
-  return R.filter(function (dir) {
-    return R.match(/v?\d+\.\d+\.\d+/, dir).length > 0;
-  }, dirs);
+function getInstalledVersions() {
+    return fs.readdirAsync(paths.PSVM_VERSIONS)
+        .then(function (dirs) {
+            return R.filter(function (dir) {
+                return R.match(/v?\d+\.\d+\.\d+/, dir).length > 0;
+            }, dirs);
+        }, function (err) {
+			if (err.code === 'ENOENT') {
```

Comment:
I don't use anything like that. I just go by the convention in the rest of the file - in the rest of this file, it looks like you're using 4 spaces everywhere, but here, it looks like you're using tabs.